Tests/SwiftLintFrameworkTests/NestingRuleTests.swift

```diff
@@ -368,4 +370,45 @@ class NestingRuleTests: SwiftLintTestCase {
 
         verifyRule(description, ruleConfiguration: ["check_nesting_in_closures_and_statements": false])
     }
+
+    func testNestingWithoutTypealiasAndAssociatedtype() {
+        var nonTriggeringExamples = NestingRule.description.nonTriggeringExamples
+        nonTriggeringExamples.append(contentsOf: ["class", "struct", "enum"].flatMap { type -> [Example] in
+            [
+                .init("""
+                    \(type) Example_0 {
+                        \(type) Example_1 {
+                            typealias Example_2_Type = Example_2.Type
+                        }
+                        \(type) Example_2 {
+
+                        }
+                    }
+                """)
```

Comment:
Let's have other test cases for `associatedtype` and `typealias` used in functions.

Reply:
I added two test cases possible to think for me